From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Martin Schmiedel <martin.schmiedel@tq-group.com>
Date: Tue, 1 Sep 2026 00:00:00 +0200
Subject: [PATCH] env: tq: k3_common: fix fdt/initrd overlap with kernel
 Image

kernel_addr_r=CONFIG_SYS_LOAD_ADDR=0x88000000, fdt_addr_r=0x8b000000
budgets only 48 MiB for the kernel Image. Armbian's MBa67xx Image
exceeds that ("FDT image overlaps OS image"), so push
fdt/fdtoverlay/initrd/ramdisk addresses up to give the Image 128 MiB
of room, keeping the original 8 MiB spacing between them.

Signed-off-by: Martin Schmiedel <martin.schmiedel@tq-group.com>
---
 include/env/tq/k3_common.env | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

--- a/include/env/tq/k3_common.env
+++ b/include/env/tq/k3_common.env
@@ -14,10 +14,10 @@
 kernel_addr_r=CONFIG_SYS_LOAD_ADDR
 scriptaddr=0x8a800000
 pxefile_addr_r=0x8ac00000
-fdt_addr_r=0x8b000000
-fdtoverlay_addr_r=0x8b800000
-initrd_addr_r=0x8c000000
-ramdisk_addr_r=0x8c000000
+fdt_addr_r=0x90000000
+fdtoverlay_addr_r=0x90800000
+initrd_addr_r=0x91000000
+ramdisk_addr_r=0x91000000
 
 /* Offset for spi_flash bootdev */
 script_offset_f=0x7f0000
-- 
2.43.0
